File: Emulator/src/main/core/net/event/MSHSReadEvent.java
package core.net.event;

import core.net.EventProducer;
import core.net.IoReadEvent;
import core.net.IoSession;
import core.net.producer.RegistryEventProducer;

import java.nio.ByteBuffer;

/**
 * The type Mshs read event.
 */
public final class MSHSReadEvent extends IoReadEvent {

    private static final EventProducer REGISTRY_PRODUCER = new RegistryEventProducer();

    /**
     * Instantiates a new Mshs read event.
     *
     * @param session the session
     * @param buffer  the buffer
     */
    public MSHSReadEvent(IoSession session, ByteBuffer buffer) {
        super(session, buffer);
    }

    @Override
    public void read(IoSession session, ByteBuffer buffer) {
        int opcode = buffer.get() & 0xFF;
        if (opcode == 14) {
            session.setProducer(REGISTRY_PRODUCER);
            session.write(true);
        }
    }

}

File: Emulator/src/main/core/net/event/MSHSReadEvent.kt
package core.net.event

import core.net.EventProducer
import core.net.IoReadEvent
import core.net.IoSession
import core.net.producer.RegistryEventProducer
import java.nio.ByteBuffer

/**
 * Handles the management server handshake read events.
 *
 * @author Emperor
 */
class MSHSReadEvent(session: IoSession, buffer: ByteBuffer) : IoReadEvent(session, buffer) {

    companion object {
        private val REGISTRY_PRODUCER: EventProducer = RegistryEventProducer()
    }

    override fun read(session: IoSession, buffer: ByteBuffer) {
        val opcode = buffer.get().toInt() and 0xFF
        if (opcode == 14) {
            session.setProducer(REGISTRY_PRODUCER)
            session.write(true)
        }
    }
}
